service/history/shardController.go

```diff
@@ -322,25 +322,38 @@ func (c *shardController) acquireShards() {
 	sw := c.metricsClient.StartTimer(metrics.HistoryShardControllerScope, metrics.AcquireShardsLatency)
 	defer sw.Stop()
 
-AcquireLoop:
+	concurrency := common.MaxInt(c.config.AcquireShardConcurrency(), 1)
+	shardActionCh := make(chan int, c.config.NumberOfShards)
+	var wg sync.WaitGroup
+	wg.Add(c.config.NumberOfShards)
```

Comment:
same comment - you only need wg of size=concurrency. And goroutines just do - defer wg.Done()

Reply:
Addressed too.